presto-main/src/main/java/com/facebook/presto/execution/SqlTask.java

```diff
@@ -422,6 +444,19 @@ public TaskInfo abortTaskResults(OutputBufferId bufferId)
         return getTaskInfo();
     }
 
+    public void removeRemoteSource(TaskId sourceTaskId)
+    {
+        requireNonNull(sourceTaskId, "sourceTaskId is null");
+
+        log.debug("Removing remote source %s from task %s", sourceTaskId, taskId);
+
+        exchangeClientManager.getExchangeClients()
+                .forEach(exchangeClient -> exchangeClient.removeRemoteSource(sourceTaskId));
+        for (ExchangeClient exchangeClient : ImmutableList.copyOf(exchangeClientManager.getExchangeClients())) {
```

Comment:
You don't need to do it twice (once with a loop and once with the `forEach`)

Reply:
Yep. Looks like I forgot to remove the loop version when addressing comments.